src/UI/Buyer/src/app/products/components/additional-image-gallery/additional-image-gallery.component.ts

```diff
@@ -16,7 +16,16 @@ export class AdditionalImageGalleryComponent {
   endIndex = this.gallerySize - 1;
   faAngleLeft = faAngleLeft;
   faAngleRight = faAngleRight;
-  constructor() {}
+  isResponsiveView: boolean;
+
+  constructor() {
+    this.isResponsiveView = window.innerWidth > 900;
+  }
+
+  @HostListener('window:resize', ['$event'])
+  onResize(event) {
+    this.isResponsiveView = event.target.innerWidth > 900;
+  }
```

Comment:
we might want to try doing this a different way. Host listener sounds dope but there is currently no way to unsubscribe from a listener which could cause a memory leak - https://github.com/angular/angular/issues/16366

That's actually the reason i updated the way i implemented my fallback url thing, I was originally using host listener.

The other thing to consider is that resize events are notorious for being performance heavy. Could we possibly solve this by just getting the current width of the window on ngOnInit and make a fair assumption that they won't be resizing it? This is aimed at mobile right? I think its a fair assumption to make.

Reply:
i take back my second point about performance. I guess its just checking a value so i don't think that should affect it but I think the unsubscribe thing might still be an issue. In that case maybe let's use observables (with the same event) so we can unsubscribe.

Reply:
Well my line of thinking is that its both for mobile and resizing the browsers. There is plenty of times when I resize my browser to fit my screen to shop, especially when comparing things. 
I update this to use an observable.